barretenberg/cpp/src/barretenberg/proof_system/circuit_builder/ultra_circuit_builder.cpp

```diff
@@ -138,6 +139,7 @@ void UltraCircuitBuilder_<Arithmetization>::create_add_gate(const add_triple_<FF
     q_lookup_type.emplace_back(0);
     q_elliptic.emplace_back(0);
     q_aux.emplace_back(0);
+    pad_additional_selectors();
```

Comment:
Does this mean we're calling a virtual function now for every gate added?

Reply:
Moved this functionality to the `Arithmetization` classes so we're more consistent in how we're achieving polymorphism